File: easy/create_counter.go
package easy

func CreateCounter(n int) func() int {
	count := n
	increment := func() int {
		current := count
		count += 1
		return current
	}
	return increment
}

func CounterCapture(nums []int, index int, counter func() int) {
	nums[index] = counter()
}

File: easy/create_counter.py
from typing import Callable


def create_counter(n: int) -> Callable[[], int]:
    count = n

    def increment() -> int:
        nonlocal count
        current = count
        count += 1
        return current

    return increment


def capture(nums: list[int], index: int, counter: Callable[[], int]) -> None:
    nums[index] = counter()
    return None
